crates/core/src/subscription/module_subscription_manager.rs

```diff
@@ -182,14 +203,27 @@ impl SubscriptionManager {
         }
     }
 
+    /// Remove any clients that have been marked for removal
+    pub fn remove_dropped_clients(&mut self) {
+        for id in self.clients.keys().copied().collect::<Vec<_>>() {
+            if let Some(client) = self.clients.get(&id) {
+                if client.dropped.load(Ordering::Relaxed) {
+                    self.remove_all_subscriptions(&id);
+                }
+            }
+        }
+    }
+
     /// Remove a single subscription for a client.
     /// This will return an error if the client does not have a subscription with the given query id.
     pub fn remove_subscription(&mut self, client_id: ClientId, query_id: ClientQueryId) -> Result<Query, DBError> {
         let subscription_id = (client_id, query_id);
         let Some(ci) = self.clients.get_mut(&client_id) else {
             return Err(anyhow::anyhow!("Client not found: {:?}", client_id).into());
         };
-
+        if ci.dropped.load(Ordering::Acquire) {
+            return Err(anyhow::anyhow!("Client not found: {:?}", client_id).into());
+        }
```

Comment:
Doesn't really matter, but `clients.get_mut(...).filter(|ci| !ci.dropped.load(Acquire))` might be a useful idiom here.